Firestore/core/test/unit/remote/serializer_test.cc

```diff
@@ -1264,6 +1266,77 @@ TEST_F(SerializerTest, EncodesMultipleFiltersOnDeeperCollections) {
   ExpectRoundTrip(model, proto);
 }
 
+TEST_F(SerializerTest, EncodesCompositeFiltersOnDeeperCollections) {
+  // (prop < 42) || (author == "ehsann" && tags array-contains "pending")
+  core::Query q =
+      Query("rooms/1/messages/10/attachments")
+          .AddingFilter(OrFilters(
+              {Filter("prop", "<", 42),
+               AndFilters({Filter("author", "==", "ehsann"),
```

Comment:
s/ehsann/cheryllin   😜

Reply:
Sorry if this wasn't clear. `s/ehsann/cheryllin` means: substitute "ehsann" with "cheryllin".
[It's the syntax for replacing text in vim](https://linuxize.com/post/vim-find-replace/#:~:text=For%20example%2C%20to%20search%20for%20the%20first%20occurrence%20of%20the%20string%20%E2%80%98foo%E2%80%99%20in%20the%20current%20line%20and%20replace%20it%20with%20%E2%80%98bar%E2%80%99%2C%20you%20would%20use%3A).